lib/connection.js

```diff
@@ -592,7 +592,7 @@ Connection.prototype.invokeCallback = function (handler, err, response) {
  * @returns {Object} The timeout object
  */
 Connection.prototype.setRequestTimeout = function (request, options) {
-  var millis = (options && options.readTimeout) || this.options.socketOptions.readTimeout;
+  var millis = (options && options.readTimeout !== undefined) ? options.readTimeout : this.options.socketOptions.readTimeout;
```

Comment:
Adapted this from clientOptions.ifUndefined to be consistent with that behavior.